src/stable/boxed.rs

```diff
@@ -591,6 +591,13 @@ impl<T, A: Allocator> Box<T, A> {
                 .1
                 .deallocate(ptr.as_non_null_ptr().cast(), Layout::new::<T>())
         };
+
+        // Override our default `Drop` implementation.
+        // Though the default `Drop` implementation drops the both the pointer and the allocator,
+        // here we only want to drop the allocator.
+        let _ = unsafe { ptr::read(&boxed.1) };
+        mem::forget(boxed);
```

Comment:
If either `deallocate` or allocators `Drop` panic this won't be called and `Box` would be dropped. This should be avoided.
To do so, first wrap this `Box` into `ManuallyDrop`, extract allocator so it would be dropped at the end of the scope, and finally deallocate the pointer.